fathom/src/surface/elaboration.rs

```diff
@@ -187,6 +187,35 @@ impl<'arena> RigidEnv<'arena> {
         env.define_prim_fun(FormatArray32, [&U32_TYPE, &FORMAT_TYPE], &FORMAT_TYPE);
         env.define_prim_fun(FormatArray64, [&U64_TYPE, &FORMAT_TYPE], &FORMAT_TYPE);
         env.define_prim_fun(FormatRepeatUntilEnd, [&FORMAT_TYPE], &FORMAT_TYPE);
+        // repeat_until_full16 : U16 -> fun (A : Format) -> (Repr A -> U16) -> Format
+        env.define_prim(
+            FormatRepeatUntilFull,
+            scope.to_scope(core::Term::FunType(
+                Span::Empty,
+                None,
+                &U16_TYPE,
+                scope.to_scope(core::Term::FunType(
+                    Span::Empty,
+                    env.name("A"),
+                    &FORMAT_TYPE,
+                    scope.to_scope(core::Term::FunType(
+                        Span::Empty,
+                        None,
+                        scope.to_scope(Term::FunType(
+                            Span::Empty,
+                            None,
+                            scope.to_scope(Term::FunApp(
+                                Span::Empty,
+                                &Term::Prim(Span::Empty, FormatRepr),
+                                &VAR0,
+                            )),
+                            &U16_TYPE,
+                        )),
+                        &FORMAT_TYPE,
+                    )),
+                )),
+            )),
+        );
```

Comment:
Writing this was a real brain-teaser but I got there eventually.

Reply:
hah, phew yeah we need a better way to handle this, sorry!